django/db/models/sql/compiler.py

```diff
@@ -146,9 +143,19 @@ def collapse_group_by(self, expressions, having):
                         getattr(expr, 'alias', None) == self.query.tables[0]):
                     pk = expr
                     break
+            # If the main model's primary key is in the query, set expressions
```

Comment:
Instead of "set expressions to that field" maybe it would be better to say, "group by that field, HAVING expressions, and ..." since it's more useful to explain what the assignment means.